pkg/compactor/block_upload.go

```diff
@@ -262,22 +269,51 @@ func (c *MultitenantCompactor) UploadBlockFile(w http.ResponseWriter, r *http.Re
 	w.WriteHeader(http.StatusOK)
 }
 
-func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) error {
+func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) {
 	level.Debug(logger).Log("msg", "completing block upload", "files", len(meta.Thanos.Files))
 
+	// start go routine that updates validation file timestamp once per minute
+	ch := make(chan string)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg, cancel)
+
+	if err := c.validateBlock(ctx, blockID, userBkt, meta); err != nil {
+		err := c.uploadValidationWithError(ctx, blockID, userBkt, err.Error())
+		if err != nil {
+			level.Error(logger).Log("msg", "error updating validation file after failed validation in object store", "blockID", blockID)
```

Comment:
note that we don't need to add "blockID" part to the log message, because `logger` is already configured in `FinishBlockUpload` to add `block` field to all log messages.